application/lib/viewmodel/log_in_screen_view_model.dart

```diff
@@ -0,0 +1,29 @@
+import 'package:flutter/material.dart';
+import '../model/user_model.dart';
+import '../service/user_database_service.dart';
+import '../utils/route_constants.dart';
+import '../utils/text_strings.dart';
+
+class LogInViewModel with ChangeNotifier {
+  void navigateToHome(BuildContext context) {
+    Navigator.pushReplacementNamed(context, homeRoute);
+  }
+
+  String? validateEmail(String value) {
+    if (!value.contains("@") && value.isNotEmpty) {
```

Comment:
Ha megfordítod az if-ek sorrendjét, akkor egyszerűbben is le tudod írni ezt a feltételt. Ha az elsőbe nem fut bele, akkor az már nem lesz igaz az alatta levőkben: 
`if (value.isEmpty) ..... 
else if !value.contains("@") ..... `